Green City UBS Delete unused endpoints Management Order Controller

LiliaMokhnatska opened the pull request:
# Green City UBS  Delete unused endpoints


## Summary Of Issue :
There are HTTP methods that are not used on the Admin Order page: Assign employee for some order, Update employee position by order, Delete Certificate, Update order address.


## Issue Link :
https://github.com/ita-social-projects/GreenCity/issues/5100

## Summary Of Changes :
Deleted:


**1.  removed endpoint "/ubs/management/update-address"**

1)removed method  updateAddressByOrderId() in ManagementOrderController
2)removed tests updateAddress(), updateOrderDetail() in ManagementOrderControllerTest
3) removed method getOrderDetailInfoDto() in ModelUtils

**2.  removed endpoint "/ubs/management/update-position-by-order"**

1)removed method  updateByOrder() in ManagementOrderController
2) removed test updatePositionByOrderTest() in ManagementOrderControllerTest
3)removed method updatePositions in UBSManagementService
4) removed methods updatePositions, collectEventsAboutUpdatingEmployeesAssignedForOrder in UBSManagementServiceImpl
5) removed tests testUpdatePositionTest(long diffParam, String eventName)testUpdatePositionThrowsOrderNotFoundException(),
testUpdatePositionThrowsPositionNotFoundException(),
testUpdatePositionThrowsEmployeeNotFoundException() in UBSManagementServiceImplTest

**3. removed  endpoint "/ubs/management/assign-employees-to-order""**

1)remove method  assignEmployeesToOrder() in ManagementOrderController
2) remove test assignEmployeeToOrder() in ManagementOrderControllerTest
3)remove method assignEmployeesWithThePositionsToTheOrder in UBSManagementService
4) remove methods assignEmployeesWithThePositionsToTheOrder, collectsEventsAboutAssigningEmployees in UBSManagementServiceImpl
5) remove tests testAssignEmployeeWithThePositionToTheOrderException
testAssignEmployeeWithThePositionToTheOrderEmployeeAlreadyExistException(),
testAssignEmployeesWithThePositionToTheOrderEmployeeNotFindException(),
testAssignEmployeesToTheOrderEmployeeNotFindException(),
testAssignEmployeesWithThePositionsNotFoundToEmployee,
testAssignEmployeesWithThePositionsToTheOrderIsNotAssignedException,
testAssignEmployeesWithThePositionsToTheOrderParams
 in UBSManagementServiceImplTest

**4. deleted unused methods and DTOs**
1) deleted methods assignEmployeeToOrderDto from

 core/src/test/java/greencity/ModelUtils.java

2) deleted methods assignEmployeesForOrderDto and assignEmployeeForOrderDto
 service/src/test/java/greencity/ModelUtils.java

3) deleted AssignEmployeesForOrderDto

4) deleted AssignForOrderEmployee



## How to test :
Steps to reproduce

1. Go to block 'Management Order Controller'
6. Click on it
7. Pay attention to the endpoints: 
Assign employee for some order,
Update employee position by order, 
Delete Certificate,
Update order address





## CHECK LIST
- [x] Code is up-to-date with the `dev` branch.
- [x] You've successfully built and run the tests locally.
- [x] There are new or updated unit tests validating the change.
- [x] JIRA/ Github Issue number & title in PR title (ISSUE-#5058: Ticket title)
- [x] This template filled (above this section).
- [ ] Sonar's report does not contain bugs, vulnerabilities, security issues, code smells ar duplication
- [x] `NEED_REVIEW` and `READY_FOR_REVIEW` labels are added.
- [x] All files reviewed before sending to reviewers

LiliaMokhnatska committed:
remove endpoint  "/ubs/management/update-address"

1)remove method  updateAddressByOrderId() in ManagementOrderController

2)remove tests updateAddress(), updateOrderDetail() in ManagementOrderControllerTest

3) remove method getOrderDetailInfoDto() in ModelUtils

remove endpoint  "/ubs/management/update-position-by-order"

1)remove method  updateByOrder() in ManagementOrderController

2) remove test updatePositionByOrderTest() in ManagementOrderControllerTest

3)remove method updatePositions in UBSManagementService

4) remove methods updatePositions, collectEventsAboutUpdatingEmployeesAssignedForOrder in UBSManagementServiceImpl

5) remove tests testUpdatePositionTest(long diffParam, String eventName)testUpdatePositionThrowsOrderNotFoundException(),
testUpdatePositionThrowsPositionNotFoundException(),
testUpdatePositionThrowsEmployeeNotFoundException() in UBSManagementServiceImplTest

remove endpoint  "/ubs/management/assign-employees-to-order""

1)remove method  assignEmployeesToOrder() in ManagementOrderController

2) remove test assignEmployeeToOrder() in ManagementOrderControllerTest

3)remove method assignEmployeesWithThePositionsToTheOrder in UBSManagementService

4) remove methods assignEmployeesWithThePositionsToTheOrder, collectsEventsAboutAssigningEmployees in UBSManagementServiceImpl

5) remove tests testAssignEmployeeWithThePositionToTheOrderException
testAssignEmployeeWithThePositionToTheOrderEmployeeAlreadyExistException(),
testAssignEmployeesWithThePositionToTheOrderEmployeeNotFindException(),
testAssignEmployeesToTheOrderEmployeeNotFindException(),
testAssignEmployeesWithThePositionsNotFoundToEmployee,
testAssignEmployeesWithThePositionsToTheOrderIsNotAssignedException,
testAssignEmployeesWithThePositionsToTheOrderParams
 in UBSManagementServiceImplTest

RomaMocherniuk commented on core/src/main/java/greencity/controller/ManagementOrderController.java:
```diff
@@ -139,29 +135,6 @@ public ResponseEntity<HttpStatus> addCertificate(
-    })
-    @PreAuthorize("@preAuthorizer.hasAuthority('EDIT_CERTIFICATE', authentication)")
-    @DeleteMapping("/deleteCertificate/{code}")
-    public ResponseEntity<HttpStatus> deleteCertificate(
```
I see that we can add `certificates`
Will we never delete them?

LiliaMokhnatska replied:
This is not in the requirements for the project. But this possibility should be left. You are right, I should leave the endpoint to remove the Certificate.

RomaMocherniuk commented on core/src/main/java/greencity/controller/ManagementOrderController.java:
```diff
@@ -371,29 +344,6 @@ public ResponseEntity<ReadAddressByOrderDto> getAddressByOrderId(
-        @ApiResponse(code = 404, message = HttpStatuses.NOT_FOUND)
-    })
-    @PreAuthorize("@preAuthorizer.hasAuthority('EDIT_ORDER', authentication)")
-    @PutMapping("/update-address")
```
What if client added wrong order addess
How will we update it?
Do we have another enpoint for it?

LiliaMokhnatska replied:
yes we have another endpoint with putch method and url /ubs/management/update-order-page-admin-info/{id}

RomaMocherniuk commented on core/src/main/java/greencity/controller/ManagementOrderController.java:
```diff
@@ -139,29 +135,6 @@ public ResponseEntity<HttpStatus> addCertificate(
-        @ApiResponse(code = 403, message = HttpStatuses.FORBIDDEN)
-    })
-    @PreAuthorize("@preAuthorizer.hasAuthority('EDIT_CERTIFICATE', authentication)")
-    @DeleteMapping("/deleteCertificate/{code}")
```
please remove this endpoint from `SecurityConfig` too

RomaMocherniuk commented on core/src/test/java/greencity/controller/ManagementOrderControllerTest.java:
```diff
@@ -337,18 +299,6 @@ void groupCoordsWithSpecifiedOnes() throws Exception {
 
-    @Test
-    void assignEmployeeToOrder() throws Exception {
-        AssignEmployeesForOrderDto assignEmployeesForOrderDto = ModelUtils.assignEmployeeToOrderDto();
```
please remove `assignEmployeeToOrderDto` too

RomaMocherniuk commented on core/src/test/java/greencity/controller/ManagementOrderControllerTest.java:
```diff
@@ -137,42 +137,12 @@ void addCertificateTest() throws Exception {
         verify(certificateService, times(1)).addCertificate(certificateDtoForAdding);
     }
 
-    @Test
```
please remove unused code from `ManagementOrderControllerTest`

RomaMocherniuk commented on service/src/test/java/greencity/service/ubs/CertificateServiceImplTest.java:
```diff
@@ -52,34 +52,6 @@ void addCertificateTest() {
         verify(certificateRepository, times(1)).save(certificate);
     }
 
-    @Test
```
please remove unused imports

LiliaMokhnatska committed:
deleted tests from UBSManagementServiceImplTest

LiliaMokhnatska committed:
deleted unused methods  and DTOs
1) deleted methods assignEmployeeToOrderDto from

 core/src/test/java/greencity/ModelUtils.java

2) deleted methods assignEmployeesForOrderDto and assignEmployeeForOrderDto
 service/src/test/java/greencity/ModelUtils.java

3) deleted AssignEmployeesForOrderDto

4) deleted AssignForOrderEmployee
